### ISSUE TITLE
Xcconfing files are generated exponentially based on number of targets and configurations

### ISSUE THREAD

**Issue by dautovicharis** [CocoaPods/CocoaPods#9966]
**Problem**: Pods xcconfig files are generated exponentially based on number of targets and configurations.

**Case**: 
- 50 targets and 3 configurations = 150 xconfig files (All same).
- All targets are using same pods versions

![Screenshot 2020-08-06 at 23 40 09](https://user-images.githubusercontent.com/7049715/89585757-b84f5400-d83e-11ea-97df-b6a61d4f4f9b.png)

![Screenshot 2020-08-06 at 23 37 39](https://user-images.githubusercontent.com/7049715/89585882-f2b8f100-d83e-11ea-9b93-999d31724da9.png)

**Desired results**:
- Want to generate only 2 xcconfig (debug/release) and use it for all targets.

I did a lot of research and tried everything, but didn't find any solution.

**Comment by dnkoutso** [comment]
Yes this is currently by design, but how does this affect you directly?

Is it a performance issue?

**Comment by dautovicharis** [comment]
I don't have a performance issue here yet.

I was only trying to find a solution for cleanup because of files with the same content.

Currently, we can survive with it, but over time this number of xcconfig files can grow up quickly by adding new project configurations.

Example: 70 targets x 10 configurations = 700 xcconfig files.

**Comment by dnkoutso** [comment]
I mean we could potentially optimize this if the contents are the same not to create a new one....I thkn that would work?

**Comment by dautovicharis** [comment]
This would be great if can be done. I'm here to help with testing or if the demo project is needed.

**Comment by dnkoutso** [comment]
No need for a demo I dont think! Feel free to try and open a PR with a fix! I'd be happy to review.

**Comment by dautovicharis** [comment]
Sounds good. I'll try -  I'll let you know. Thanks
